section1/reflection.md

```diff
@@ -2,18 +2,43 @@
 
 1. Regarding the blog posts in Part A, how do you feel about asking questions? Do you tend to ask them too soon, or wait too long, or somewhere in between?
 
+I thought I waited too long to ask questions but one thing I realized is the difference between productive and unproductive problem-solving. As long as I'm utilizing the tools at my disposal, like Google, lecture/my notes, Slack search, and feel that I'm making meaningful progress in finding an answer, that is productive problem-solving. It crosses into unproductive territory when I continue to hit an error message trying all of the solutions I have found. According to the articles and Turing's guidance, I should spend no more than 20-30 minutes in this unproductive space. Deciding when something becomes unproductive is what I'm still figuring out. I'm getting there though!
+
 2. What are the data types you learned about in this section? In your own words, define each.
+I learned about three primitive (fundamental) data types:
+  * String: letters (words), characters, and numbers wrapped in quotes
+  * Number: all positive and negative numbers including whole numbers and floating point numbers (decimals
+  * Boolean: true or false
 
 3. How would you log the string `"Hello World!"` to the console?
-
+```javaScript
+console.log(`Hello World!`);
+```
 4. What is/are the character(s) you would use to indicate comments in a JavaScript file? What is the purpose of a code comment?
+``// for a single-line comment``
+
+```javaScript
+/* for a
+multi-line comment
+like this one */
+```
 
 5. In your own words, what is a variable? How would you explain it to a 5 year old?
+A *variable* is a temporarily stored bit of data needed to do a job. It can change each time a script runs.
+A variable is like a plate. Your plate can be different colors and sizes just like the kind of food it holds can be different too. Some plates are better for eating some foods versus others, so you pick the best plate for the kind of food you want to eat!
 
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
+  * username: script
+  * songLength: number
+  * clickPlayButton: boolean
 
 7. In your own words, explain what concatenation is.
+**Concatenation** is joining together separate variables with an operator, typically **+**.
 
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
+  A welcome message telling me where I left off in my typing lesson.
 
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+~~I have no questions.~~
+  * How do reassigned variables work? I did the exercises but do not fully understand how Javascript interprets that the variable should be reassigned if you are just retyping the variable name and assigning it a new name. I will research this independently as well.
```

Comment:
`the variable should be reassigned` is actually the same thing as `you are just retyping the variable name and assigning it a new ~~name~~ value` - so I think you maybe already answered your question! 

In a physical world, I could think of this as food in a pyrex container. Today, I have leftover ravioli in the container. Once I eat it, I will refill that same container with leftover tacos. The container (variable name) stayed the same, but the value (food) changed.